File: Software/Cloty/scripts/gen-ubicacion-seed.py
import json
from pathlib import Path

base = Path(__file__).resolve().parent.parent / "cloty-api" / "src" / "main" / "resources" / "data"
regions = json.loads((base / "chile-regiones.json").read_text(encoding="utf-8"))
comunas = json.loads((base / "chile-comunas.json").read_text(encoding="utf-8"))
out = Path(__file__).resolve().parent.parent / "cloty-ubicacion-seed.sql"

lines = [
    "-- Seed regiones y comunas de Chile (CUT SUBDERE)",
    "-- Ejecutar después de cloty-mysql.sql:",
    "--   mysql -u root -p cloty < cloty-ubicacion-seed.sql",
    "SET NAMES utf8mb4;",
    "",
    "INSERT INTO region (codigo_region, nombre) VALUES",
]
lines.append(",\n".join(
    f"  ('{r['codigo_region']}', '{r['nombre_region'].replace(chr(39), chr(39) + chr(39))}')"
    for r in regions
) + ";")
lines.append("")
lines.append("INSERT INTO comuna (codigo_comuna, codigo_region, nombre) VALUES")
rows = []
for c in comunas:
    nombre = c["nombre_comuna"].replace("'", "''")
    rows.append(f"  ('{c['codigo_comuna']}', '{c['codigo_region']}', '{nombre}')")
lines.append(",\n".join(rows) + ";")
out.write_text("\n".join(lines), encoding="utf-8")
print(f"Wrote {out} ({len(regions)} regions, {len(comunas)} comunas)")

File: Software/Cloty/scripts/gen-ubicacion-seed.js
const fs = require('fs');
const path = require('path');

const base = path.join(__dirname, '..', 'cloty-api', 'src', 'main', 'resources', 'data');
const regions = JSON.parse(fs.readFileSync(path.join(base, 'chile-regiones.json'), 'utf8'));
const comunas = JSON.parse(fs.readFileSync(path.join(base, 'chile-comunas.json'), 'utf8'));
const out = path.join(__dirname, '..', 'cloty-ubicacion-seed.sql');

const esc = (s) => s.replace(/'/g, "''");
const lines = [
  '-- Seed regiones y comunas de Chile (CUT SUBDERE)',
  '-- Ejecutar después de cloty-mysql.sql:',
  '--   mysql -u root -p cloty < cloty-ubicacion-seed.sql',
  'SET NAMES utf8mb4;',
  '',
  'INSERT INTO region (codigo_region, nombre) VALUES',
  regions.map((r) => `  ('${r.codigo_region}', '${esc(r.nombre_region)}')`).join(',\n') + ';',
  '',
  'INSERT INTO comuna (codigo_comuna, codigo_region, nombre) VALUES',
  comunas.map((c) => `  ('${c.codigo_comuna}', '${c.codigo_region}', '${esc(c.nombre_comuna)}')`).join(',\n') + ';',
];
fs.writeFileSync(out, lines.join('\n'), 'utf8');
console.log(`Wrote ${out} (${regions.length} regions, ${comunas.length} comunas)`);
